src/engine/TransitivePathImpl.h

```diff
@@ -232,15 +226,21 @@ class TransitivePathImpl : public TransitivePathBase {
    */
   CPP_template(typename Node)(requires ql::ranges::range<Node>) NodeGenerator
       transitiveHull(const T& edges, LocalVocab edgesVocab, Node startNodes,
-                     std::optional<Id> target, bool yieldOnce) const {
+                     TripleComponent target, bool yieldOnce) const {
     ad_utility::Timer timer{ad_utility::Timer::Stopped};
+    LocalVocab targetHelper;
+    std::optional<Id> targetId =
+        target.isVariable()
+            ? std::nullopt
+            : std::optional{std::move(target).toValueId(
+                  _executionContext->getIndex().getVocab(), targetHelper)};
```

Comment:
Comment that the `target` is only used for comparisons and never stored in the result.